modules/wordads/php/widgets.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Widget for inserting an ad into your sidebar
+ *
+ * @since 0.1
+ * TODO: sanitize all the output
+ */
+class WordAds_Sidebar_Widget extends WP_Widget {
+
+	private static $allowed_tags = array( 'mrec', 'wideskyscraper' );
+
+	private $options = array();
+
+	function __construct() {
+		parent::__construct(
+			'wordads_sidebar_widget',
+			__( 'WordAds Widget (Jetpack)', 'jetpack' ),
+			array( 'description' => __( 'Insert a WordAds ad wherever you can place a widget.', 'jetpack' ) )
```

Comment:
@eliorivero Okay, so I played around with this a bunch and I don't think this widget is going to be a good fit for selective refresh. Namely, what we do is load a div with a JS snippet which grabs the ad code and inserts it via document.write(). Unfortunately, this seems to generate `Error: document.write() is forbidden` as this action is not permitted.

Perhaps there's a way to allow access to document.write()? Otherwise, I think it's best to let the page reload when a new unit is selected.